inspirehep/dojson/hep/fields/bd0xx.py

```diff
@@ -0,0 +1,331 @@
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""DoJSON rules for MARC fields in 0xx."""
+
+from __future__ import absolute_import, division, print_function
+
+import re
+
+import pycountry
+
+from dojson import utils
+from idutils import is_doi, is_handle, normalize_doi
+
+from inspire_schemas.utils import load_schema
+from inspirehep.utils.helpers import force_force_list
+
+from ..model import hep, hep2marc
+from ...utils import force_single_element
+
+
+RE_LANGUAGE = re.compile('\/| or | and |,|=|\s+')
+
+
+@hep.over('isbns', '^020..')
+@utils.for_each_value
+def isbns(self, key, value):
+    def _get_medium(value):
+        def _normalize(medium):
+            schema = load_schema('hep')
+            valid_mediums = schema['properties']['isbns']['items']['properties']['medium']['enum']
+
+            medium = medium.lower().replace('-', '').replace(' ', '')
+            if medium in valid_mediums:
+                return medium
+            elif medium == 'ebook':
+                return 'online'
+            elif medium == 'paperback':
+                return 'softcover'
+
+            return ''
+
+        medium = force_single_element(value.get('b', ''))
+        normalized_medium = _normalize(medium)
+
+        return normalized_medium
+
+    def _get_isbn(value):
+        def _normalize(isbn):
+            return isbn.lower().replace('-', '')
+
+        isbn = force_single_element(value.get('a', ''))
+        normalized_isbn = _normalize(isbn)
+
+        return normalized_isbn
+
+    return {
+        'medium': _get_medium(value),
+        'value': _get_isbn(value),
+    }
+
+
+@hep2marc.over('020', 'isbns')
+@utils.for_each_value
+def isbns2marc(self, key, value):
+    return {
+        'a': value.get('value'),
+        'b': value.get('medium'),
+    }
+
+
+@hep.over('dois', '^0247.')
+def dois(self, key, value):
+    """Populate the ``dois`` key.
+
+    Also populates the ``persistent_identifiers`` key through side effects.
+    """
+    def _get_first_non_curator_source(sources):
+        sources_without_curator = [el for el in sources if el.upper != 'CURATOR']
+        return force_single_element(sources_without_curator)
+
+    def _is_doi(id_, type_):
+        return (not type_ or type_.upper() == 'DOI') and is_doi(id_)
+
+    def _is_handle(id_, type_):
+        return (not type_ or type_.upper() in ('DOI', 'HDL')) and is_handle(id_)
```

Comment:
Because it is the reality on legacy. I guess this could be simplified by cleaning up legacy. Ha! It's only 30 records:
https://inspirehep.net/search?ln=en&ln=en&p=0247%3ADOI+-0247_a%3A10.*&of=hb&action_search=Search&sf=earliestdate&so=d&rm=&rg=25&sc=0&wl=0